Src/Base/AMReX_ParmParse.H

```diff
@@ -1612,6 +1616,7 @@ public:
         // vector<vector<std::string>>.
         std::vector<std::vector<std::string>> m_vals;
         mutable Long                          m_count = 0;
+        std::variant<std::string, bool, int, long, long long, amrex::IntVect, amrex::Box, float, double> m_typehint = std::string{};
```

Comment:
Maybe we could make it cheaper by using `T*` instead of T.

Reply:
Good point(er), hehe

Reply:
Updated :)